sway/commands/move.c

```diff
@@ -550,17 +549,23 @@ static struct cmd_results *cmd_move_container(bool no_auto_back_and_forth,
 		break;
 	}
 
-	// restore focus on destination output back to its last active workspace
-	struct sway_workspace *new_workspace =
-		output_get_active_workspace(new_output);
-	if (!sway_assert(new_workspace, "Expected output to have a workspace")) {
-		return cmd_results_new(CMD_FAILURE,
-				"Expected output to have a workspace");
-	}
-	if (new_output_last_ws && new_output_last_ws != new_workspace) {
-		struct sway_node *new_output_last_focus =
-			seat_get_focus_inactive(seat, &new_output_last_ws->node);
-		seat_set_raw_focus(seat, new_output_last_focus);
+	struct sway_output *new_output = node_get_output(destination);
+	struct sway_workspace *new_output_last_ws = NULL;
+	if (new_output && old_output == new_output) {
+		new_output_last_ws = output_get_active_workspace(new_output);
```

Comment:
This needs to be before the `container_move_to_{workspace,container}` calls. If the container is focused, the active workspace will be changed and this value won't be correct